### ISSUE TITLE
Username and user ID (in AwfulPreferences) are unreliable

### ISSUE THREAD

**Issue by baka-kaba** [Awful/Awful.apk#505]
The userID field is only assigned when you're on probation, parsed from the probation message link. So it's in the AwfulPreferences object, but it's probably completely meaningless. Right now it's only used for people on probation (a link to the LC) but if anyone tries to use it for anything else it'll be a bit of a problem

The username field is entered by the user in the settings and could be a typo / outright lie (I'm sure people in threads have shown up in Crashlytics under a different name). Trouble is it's used to identify things like your posts and threads - it actually sets the 'can open/close this thread' flag when parsing the thread list - and it could cause some weird behaviour

Anyway it would be better to scrape these automatically, preferably as a ``SyncManager`` task when the app starts up (in case something's changed, probably your username if anything)

**Comment by Sereri** [comment]
As far as the username is concerned, I think the user input doesn't actually stick at the moment. I think I remember trying to change it and it not working. We should probably fix it in this case, or, well, remove the feature. I don't think anyone uses it anyway.

**Comment by baka-kaba** [comment]
Yeah you're right, you can see where it's used and updated if you do Find Usages on ``Keys.USERNAME``, looks like it's automatically updated when you log in, and also if you parse forum icons. So you can temporarily break it if you change your username, but it resets when you log in or when the forums are updated. I reckon it would be better to explicitly update that

But yeah, in that case there's really no need to let people set it themselves huh. Just causes trouble
